mitodl/mit-open
# Signup Popover

### What are the relevant tickets?
Closes https://github.com/mitodl/hq/issues/4591

### Description (What does it do?)
Adds a signup popover to
- channel page "Subscribe" button
- Resource card "action" buttons [TODO]

### Screenshots (if appropriate):
<img width="1720" alt="Screenshot 2024-06-18 at 5 05 03 PM" src="https://github.com/mitodl/mit-open/assets/9010790/bec8b41a-228c-4c3c-b3ab-b6444d5c674d">
<img width="664" alt="Screenshot 2024-06-18 at 5 05 15 PM" src="https://github.com/mitodl/mit-open/assets/9010790/468b2209-03f3-481d-a8c9-823dd3c0db28">


<img width="1724" alt="Screenshot 2024-06-18 at 4 52 37 PM" src="https://github.com/mitodl/mit-open/assets/9010790/e08056c1-faab-4f78-aa65-b699cea4ff0d">
<img width="502" alt="Screenshot 2024-06-18 at 4 31 21 PM" src="https://github.com/mitodl/mit-open/assets/9010790/6390d783-780c-4ab7-ad33-e23d1175bafa">

<img width="1627" alt="Screenshot 2024-06-18 at 4 52 52 PM" src="https://github.com/mitodl/mit-open/assets/9010790/541e0175-f218-4e02-9989-7c21e79f9a0f">
<img width="578" alt="Screenshot 2024-06-18 at 4 53 51 PM" src="https://github.com/mitodl/mit-open/assets/9010790/a114fa6a-7e20-4c6c-97fd-fa2978c23437">


### How can this be tested?
1. Check Cards on homepage, search page, and channel pages:
    - Click the "Add to list" buttons as an authenticated user (staff or otherwise). There should be no changes:
        - Bookmark = "Add to user list"
        - Itemized list = "Add to learning path"
     - As an unauthenticated user, you should only see the bookmark button, and it should open a popover prompting signup.
2. Check "Subscribe" button on channel pages, e.g., http://localhost:8063/c/unit/ocw
     - authenticated users: Subscribe / unsubscribe functions as before
     - unauthenticated users get the popover for signup.
3. http://localhost:6006/?path=/story/smoot-design-popover--basic and the adjacent two popover stories are worht looking at i guess.
